### PR TITLE
Addition of a Hinge2JointWithBacklash PROTO

### PR DESCRIPTION
**Pull request by ad-daniel**
**Description**
This PR introduces the modeling of backlash for Hinge2Joints.

**Tasks**
Add the list of tasks of this PR.
  - [x] implement PROTO
    - [x] proto image
  - [x] implement sample world
  - [x] implement unit test
  - [x] update documentation
     - [x] Hinge2JointWithBacklash documentation
     - [x] sample world documentation
  - [x] cleanup
    - [x] world info/description, header comments, ...
    - [x] simplify makefiles
    - [x] .gitignore binaries
    - [x] verify code style
- [x] changelog

**Screenshots**

https://user-images.githubusercontent.com/44834743/111245480-fbf41280-8604-11eb-8d52-509109849f83.mp4

### CODE REVIEW

File: docs/guide/hinge-2-joint-with-backlash.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+If set to zero, the limiter in this direction is removed and therefore the corresponding fields `gearMass` and `outputSensor` are ignored.
+If both `backlash` and `backlash2` are set to zero, `startPoint` is also ignored and the [PROTO](../reference/proto.md) reverts to a classic [Hinge2Joint](../reference/hinge2joint.md).
+
+- `backlash2`: this field specifies the amount of clearance in the transmission associated to `axis2` of the [Hinge2Joint](../reference/hinge2joint.md).
+If set to zero, the limiter in this direction is removed and therefore the corresponding fields `gearMass2` and `outputSensor2` are ignored.
+If both `backlash` and `backlash2` are set to zero, `startPoint` is also ignored and the [PROTO](../reference/proto.md) reverts to a classic [Hinge2Joint](../reference/hinge2joint.md).
+
+- `gearMass`: this field specifies the mass of the transmission. If both `backlash` and `backlash2` are defined, then this mass is split among both axes, otherwise it's entirely assigned to the axis with non-zero backlash.
```
Comment (omichel) [inline docs/guide/hinge-2-joint-with-backlash.md:45]: Why not having `gearMass` and `gearMass2`? That would be more capable and easier to understand.

File: tests/protos/controllers/hinge_2_joint_with_backlash/hinge_2_joint_with_backlash.c [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  WbDeviceTag reference_hinge2joint_sensor2 = wb_robot_get_device("reference_hinge2joint_sensor2");
+  wb_position_sensor_enable(reference_hinge2joint_sensor, TIME_STEP);
+  wb_position_sensor_enable(reference_hinge2joint_sensor2, TIME_STEP);
+
+  while (wb_robot_step(TIME_STEP) != -1.0) {
+    if (velocity_motor == 0.0 && velocity_motor2 != 0.0) {  // actuating only axis2
+      reference_hinge2joint_position = wb_position_sensor_get_value(reference_hinge2joint_sensor2);
+    } else {
+      reference_hinge2joint_position = wb_position_sensor_get_value(reference_hinge2joint_sensor);
+    }
```
Comment (omichel) [inline tests/protos/controllers/hinge_2_joint_with_backlash/hinge_2_joint_with_backlash.c:53]: See [#CS360 Single if, while or for statement must be written without brackets](https://github.com/cyberbotics/webots/wiki/CPP-Coding-Style#cs360-single-if-while-or-for-statement-must-be-written-without-brackets-the-statement-part-should-be-put-on-a-separate-line-in-case-of-nested-block-keep-brackets):
```suggestion
    if (velocity_motor == 0.0 && velocity_motor2 != 0.0)  // actuating only axis2
      reference_hinge2joint_position = wb_position_sensor_get_value(reference_hinge2joint_sensor2);
    else
      reference_hinge2joint_position = wb_position_sensor_get_value(reference_hinge2joint_sensor);
```

File: tests/protos/controllers/hinge_2_joint_with_backlash/hinge_2_joint_with_backlash.c [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+    if (reference_hinge2joint_position > ANGLE_LIMIT) {
+      for (int i = 0; i < 5; ++i) {
+        if (velocity_motor == 0.0 && velocity_motor2 != 0.0) {
+          wb_motor_set_velocity(motor2[i], -velocity_motor2);  // first test, only axis2 actuated
+        } else {
+          wb_motor_set_velocity(motor[i], -velocity_motor);  // second test, only axis actuated
+        }
```
Comment (omichel) [inline tests/protos/controllers/hinge_2_joint_with_backlash/hinge_2_joint_with_backlash.c:61]:
```suggestion
        if (velocity_motor == 0.0 && velocity_motor2 != 0.0)
          wb_motor_set_velocity(motor2[i], -velocity_motor2);  // first test, only axis2 actuated
        else
          wb_motor_set_velocity(motor[i], -velocity_motor);  // second test, only axis actuated
```

File: tests/protos/controllers/hinge_2_joint_with_backlash/hinge_2_joint_with_backlash.c [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+      }
+    }
+    if (reference_hinge2joint_position < -ANGLE_LIMIT) {
+      for (int i = 0; i < 5; ++i) {
+        if (velocity_motor == 0.0 && velocity_motor2 != 0.0) {
+          wb_motor_set_velocity(motor2[i], velocity_motor2);  // first test, only axis2 actuated
+        } else {
+          wb_motor_set_velocity(motor[i], velocity_motor);  // second test, only axis actuated
+        }
```
Comment (omichel) [inline tests/protos/controllers/hinge_2_joint_with_backlash/hinge_2_joint_with_backlash.c:70]:
```suggestion
        if (velocity_motor == 0.0 && velocity_motor2 != 0.0)
          wb_motor_set_velocity(motor2[i], velocity_motor2);  // first test, only axis2 actuated
        else
          wb_motor_set_velocity(motor[i], velocity_motor);  // second test, only axis actuated
```

**Review by omichel — CHANGES_REQUESTED** [review]
It looks very good.

File: docs/guide/hinge-2-joint-with-backlash.md (hunk 1 above)
Comment (ad-daniel) [reply docs/guide/hinge-2-joint-with-backlash.md]: You're right, it's more intuitive anyway.

**Review by omichel — APPROVED** [review]
Thank you.

**Pull request merged**